### PR TITLE
Added PtRatio and looseId

### PR DESCRIPTION
**Pull request by GLP90**

### CODE REVIEW

File: python/common_variables_cff.py [hunk 1]
```diff
@@ -55,6 +55,8 @@
     emS9Energy  = cms.string("calEnergy.emS9"),
 )
 TrackQualityVariables = cms.PSet(
+
+    #dz	 	= cms.string("abs(bestTrackRef.dz(bestTrackRef.referencePoint))"),
```
Comment (gpetruc) [inline python/common_variables_cff.py]: can you remove these two useless lines?

**Comment by gpetruc** [comment]
Hi Gael,

For the AddPtRatio module:
- you should add a configurable parameter "maxDeltaR"  specifying the maximum &Delta;R for the matching (and the standard configuration should have 0.4)
- there's a `deltaR` function in CMSSW (defined in `DataFormats/Math/interface/deltaR.h`) that takes two candidates and gives the &Delta;R, so you can simplify your code removing the TVector3 etc.

Also, you should get the jets with the JEC applied. I'd suggest to add the producer for corrected jets in the `common_modules.cff`.

Giovanni

**Comment by GLP90** [comment]
Hello Giovanni,

Thank you for the tips and comments. I implemented the first two bullets. I didn't fully understand your last suggestion. Do you mean that the PtRatio should be run on another set of jets than the ak4PFJetsCHS ? Could you tell what name corresponds to the jets with the corrected energy ?

Gael

**Comment by gpetruc** [comment]
Hi,

Corrected jets do not exist already in the AOD (at least, I don't think
they do), so you need to run some EDProducer from JetMET to take the
ak4PFJetsCHS and apply the jet energy corrections.
It should be documented on the JetMET twikipages

   Giovanni

**Comment by perrozzi** [comment]
ciao, I superseeded this pull request
https://github.com/cms-analysis/MuonAnalysis-TagAndProbe/compare/74X...perrozzi:GLP90-local_tp
including also ptrel and L1L2L3 jec for ak4PFJetsCHS.
the commits are done on top of the mini isolation ones, so just after you merge that pr https://github.com/cms-analysis/MuonAnalysis-TagAndProbe/pull/8 I create a new one

**Comment by gpetruc** [comment]
Why do you need all these lines of code and shifting back and forth from the 4 coordinates as float to 4-vector classes and vice-versa just to compute ptRel? e.g. in cmgtools it's only 4 lines https://github.com/CERN-PH-CMG/cmg-cmssw/blob/CMGTools-from-CMSSW_7_2_3/CMGTools/TTHAnalysis/python/analyzers/ntupleTypes.py#L215-L219 and probably even that contains some unneeded code (e.g. the Vect())

More important: after some discussion in the susy group, we didn't yet settle on how to apply jet energy corrections, i.e. which corrections levels, before vs after subtracting the lepton, etc.  So, it's probably better to keep this on hold until we decide or at least until we narrow it to very few options. @mmarionncern @cbotta can keep you updated on this while I'm away.

**Comment by perrozzi** [comment]
All these lines come from some recycled some code, I will shrink it to the minimum needed. I'll wait for the updates on the jet collection and corrections to use before to create a PR

**Pull request closed without merging**